pyqtorch/primitive.py

```diff
@@ -57,9 +57,19 @@ def unitary(self, values: dict[str, Tensor] | Tensor = dict()) -> Tensor:
         return self.pauli.unsqueeze(2)
 
     def forward(self, state: Tensor, values: dict[str, Tensor] | Tensor = dict()) -> Tensor:
-        return apply_operator(
-            state, self.unitary(values), self.qubit_support, len(state.size()) - 1
-        )
+        if isinstance(state, DensityMatrix):
+            # error type int | tuple[int, ...] expected "int" but for now it hande only 1-qubit gate
+            return DensityMatrix(
```

Comment:
i would prefer to have separate `_forward_statevec` and `_forward_densitymat` methods which you then call in the forward

Reply:
Ok sure